### PR TITLE
tests: add test for filtering+paging+aggregation

### PR DESCRIPTION
**Pull request by psarna**
The test case ensures that filtering with paging works fine if
an aggretation function is used - the results should be presented
as a single aggregated value, without partial results per-page.

Ref #4540

### CODE REVIEW

**Comment by psarna** [comment]
@nyh 
1. as I mentioned in #4540, I think that it might have accidentally got fixed by a different series.
2. Page size in CQL is counted in the number of rows. There's also a max query limit (1MB) and hitting that limit results in returning a smaller page of results. My tests use custom query_options, in which I pass a custom page_size parameter (2 and 3), this line is the example of page size 2:
~~~
cql3::query_options::specific_options{2, nullptr, {}, api::new_timestamp()});
~~~
You can also verify with my previous tests, e.g. `test_allow_filtering_limit`, which use a similar technique to check for different paging corner cases.

File: tests/filtering_test.cc [hunk 1]
```diff
@@ -1153,3 +1153,27 @@ SEASTAR_TEST_CASE(test_filtering) {
 
     });
 }
+
+SEASTAR_TEST_CASE(test_filtering_paging_and_aggregation) {
+    return do_with_cql_env_thread([] (cql_test_env& e) {
+        e.execute_cql("CREATE TABLE fpa (id int primary key, v int)").get();
+        for (int i = 0; i < 1000; ++i) {
+            e.execute_cql(format("INSERT INTO fpa (id, v) VALUES ({}, {})", i, i % 2)).get();
```
Comment (dekimir) [inline tests/filtering_test.cc]: (In the future,) Please consider using [`cquery_nofail`](https://github.com/scylladb/scylla/blob/master/tests/cql_assertions.hh#L82), which accurately reports failure location.  This is especially valuable for loop bodies like this.

Comment (psarna) [reply tests/filtering_test.cc]: Neat. I've already done that for my newest series and I'll switch to `cquery_nofail` in the future. Thanks!
